Add tref

HuiyaChuang-NOAA opened the pull request:
Add IO of foundation temperature as well as fix how sea ice grid point is identified as land now that future coupled GFS will use fractional ice.

Commit:
Add foundation temperature per request from coupling group

Commit:
add founfation T to inline post control file

Commit:
add founfation T to inline post control file flux file part instead

Commit:
Change sea ice threshold for identifying sea ice from .1 to 0

Commit:
Resolved merge conflict by removing RQSTFLD.F

WenMeng-NOAA commented on sorc/ncep_post.fd/FIXED.f:
```diff
@@ -109,7 +110,7 @@ SUBROUTINE FIXED
              GRID1(I,J) = SPVAL
               IF(SM(I,J)   /= SPVAL) GRID1(I,J) = 1. - SM(I,J)
-              IF(SICE(I,J) /= SPVAL .AND. SICE(I,J) > 0.1) GRID1(I,J) = 0.
+              IF(SICE(I,J) /= SPVAL .AND. SICE(I,J) > 0.0) GRID1(I,J) = 0.
```
@HuiyaChuang-NOAA The UPP regression testing indicates this change having impact on rap/hrrr/3drtma land-sea mask. Would you consider to add condition as:
IF (modelname = "GSF" or modelname == "FV3R") then
   if(sice(i,j)/=spaval .and. sice(i,j) > 0.) grid1(i,j)=0.
else
   if(sice(i,j)/=spavel .and. sice(i,j) > 0.1) grid1(i,j)=0.
endif

HuiyaChuang-NOAA replied:
@WenMeng-NOAA are you seeing the land mask in GFS?  If rap/hrrr/rtma uses binary sea ice, you shouldn't see a change in land mask unless sice is a very small number instead of zero for a grid point without sea ice.  I am thinking using 
"SMALL=1.E-6" instead of zero.  Are you willing to give this another try?  If you really need to get back to your wcoss2 transition work, I will make the changes as you suggested.

HuiyaChuang-NOAA replied:
@WenMeng-NOAA made the update per your suggestion.  Ran a test to ensure new land mask and foundation T
looked fine.  Just committed the changes to my branch.

WenMeng-NOAA replied:
@HuiyaChuang-NOAA I don't see recent changes in your branch Add_tref.

HuiyaChuang-NOAA replied:
@WenMeng-NOAA sorry only made local commit but just pushed it to repo

WenMeng-NOAA replied:
@HuiyaChuang-NOAA Now it looks good to me. I will conduct regression tests. Thanks!

Commit:
made updates to only use new ice fraction threshold to identify sea
ice grid point for GFS and RRFS so that legacy models land masks
won't be affected by this change.

WenMeng-NOAA commented:
@LydiaStefanova-NOAA Please let me know if you have comments/suggestion on this PR.

LydiaStefanova-NOAA commented:
@WenMeng-NOAA Could you please point me to a sample output file you get after these changes?

WenMeng-NOAA commented:
@LydiaStefanova-NOAA See /scratch1/NCEPDEV/stmp2/Wen.Meng/gfs.20210216/gfs.t00z.sfluxgrbf006.grib2 for foundation temperature. Our regression test case has no land-sea mask issue. You should see land-sea mask fixing from Huiya's test case.

WenMeng-NOAA commented:
The UPP regression tests were conducted on WCOSS-Dellm WCOSS-Cray and Hera. With this PR merged, the baseline for GFS needs to be recreated.

LydiaStefanova-NOAA commented:
Thanks, @WenMeng-NOAA - I can't comment on the code side of things, but checked the output and confirm that foundation temperature is present, and that the land/sea mask and ice concentrations are in agreement. Thank you very much for these fixes!

WenMeng-NOAA commented:
@LydiaStefanova-NOAA Thanks for confirming.

WenMeng-NOAA commented:
This PR is ready for merging.

HuiyaChuang-NOAA commented:
@WenMeng-NOAA thank you for running UPP regression tests and merging the code quickly.

@LydiaStefanova-NOAA @JessicaMeixner-NOAA your requested changes for land masks and foundation T output are now in
stand-alone post.  The current plan is to have these changes in inline post in Feb.

JessicaMeixner-NOAA commented:
Thanks @WenMeng-NOAA and @HuiyaChuang-NOAA for these updates, we look forward to being able to use them in the inline post in the future.
